apiserver/facades/client/charms/charmhubrepo.go

```diff
@@ -271,12 +271,12 @@ func (c *chRepo) selectNextBase(bases []transport.Base, origin corecharm.Origin)
 		break
 	}
 	if !found {
-		return corecharm.Platform{}, errors.NotFoundf("base")
+		return corecharm.Platform{}, errors.NotFoundf("no bases located for architecture %q", origin.Platform.Architecture)
```

Comment:
```suggestion
		return corecharm.Platform{}, errors.NotFoundf("bases matching architecture %q", origin.Platform.Architecture)
```
Then it reads better